src/Main.elm

```diff
@@ -1,16 +1,265 @@
 module Main exposing (main)
 
-import Html exposing (Html)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Html.App as Html
+import Result.Extra as Result
+import Hierarchy as H exposing (Hierarchy)
+import Data.Api.Endpoints as Api
 
 
--- Import Modules
+-- MODEL
 
-import Data.Audience
-import Data.AudienceFolder
+
+buildInitialHierarchy : Result String Hierarchy
+buildInitialHierarchy =
+    let
+        folders =
+            Api.audience_folders
+
+        audiences =
+            Api.audiences
+
+        buildHierarchy folders audiences =
+            let
+                withFolders =
+                    List.foldl
+                        (\a b -> b `Result.andThen` (H.registerFolder a))
+                        (Ok H.empty)
+                        folders
+            in
+                List.foldl
+                    (\a b -> b `Result.andThen` (H.registerFile a))
+                    withFolders
+                    audiences
+    in
+        folders `Result.andThen` \a -> audiences `Result.andThen` \b -> buildHierarchy a b
+
+
+type Model
+    = Error String
+    | Loaded (Maybe H.Node) Hierarchy
+
+
+init : Model
+init =
+    Result.unpack Error (Loaded Nothing) buildInitialHierarchy
+
+
+
+-- UPDATE
+
+
+type Msg
+    = GoDown Int
+    | GoUp Int
```

Comment:
here it starts to be a bit confusing...
- GoDown Int -> Int is id, right?
- GoUp Int -> Int is number of levels up right?

Quick solution can be to add type alias to Int like `type alias Id = Int` and use it in case of `GoDown`. Not bullet proof but better I think.